test/integration/array_test.go

```diff
@@ -57,8 +58,8 @@ func TestArray(t *testing.T) {
 
 	t.Run("concurrent array add/delete simple test", func(t *testing.T) {
 		ctx := context.Background()
-
-		d1 := document.New(key.Key(t.Name()))
+		testDocumentKey := "concurrent-array-add-simple"
```

Comment:
We used `t.Name` to generate a unique document key for each test and used it to prevent documents created in different tests from interfering with each other.

If we manually generate unique names per test, we don't know if they interfere with each other or not.

How about adding a function in `helper` package that changes the name generated by t.Name to valid name of key.Key?

https://github.com/yorkie-team/yorkie/blob/main/test/helper/helper.go#L75

Reply:
Using the test name as the document key seems more inappropriate. To match the current rules, we need to recreate a new function.

In this case, eventually, the test code name must be uniquely matched to the document key rule.

It takes more dependencies.

I think the test name itself has nothing to do with the document key.

Reply:
I thought the helper function was something like the below:

The helper function appears to be designed to return a valid document key for testing purposes. The code takes the current test's name and converts it into a string that is safe to use as a document key. The resulting string is lowercase, alphanumeric, and any non-alphanumeric characters are replaced with hyphens.

```go
// TestDocKey returns the valid document key for test.
func TestDocKey(t *testing.T) string {
	name := t.Name()

	if len(name) < 3 {
		log.Fatal("test name should be longer than 3")
	}

	if len(name) > 60 {
		name = name[:60]
	}

	sb := strings.Builder{}
	for _, c := range name {
		if c >= 'A' && c <= 'Z' {
			sb.WriteRune(c + ('a' - 'A'))
		} else if c >= 'a' && c <= 'z' {
			sb.WriteRune(c)
		} else if c >= '0' && c <= '9' {
			sb.WriteRune(c)
		} else {
			sb.WriteRune('-')
		}
	}

	return sb.String()
}

// Replace `t.Name()` with `helper.TestDocKey(t)` in tests
// before: `TestProjectAuthWebhook/authorization_webhook_test`
// after: `testprojectauthwebhook-authorization-webhook-test`
doc := document.New(key.Key(helper.TestDocKey(t)))
```

Reply:
I added help.TestDocumentKey and help.BenchDocumentKey function. 

help.TestDocumentKey return automatically unique valid document key by t.Name()